pkg/cmd/roachtest/tests/hardware_limiter.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+	"github.com/dustin/go-humanize"
+)
+
+// hardwareLimiter is used to limit the performance of hardware in different tests.
+type hardwareLimiter struct {
+	context context.Context
+	test    test.Test
```

Comment:
Similarly I think it is better to take `test` and in fact you need a `Logger` only I think